doc: Updated release notes for Matter (#10464)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

      See :ref:`matter_lock_sample_wifi_thread_switching` in the sample documentation for more information.
    * Wi-Fi low power configuration using Wi-Fi's :ref:`Legacy Power Save mode <ug_nrf70_developing_powersave_dtim_unicast>`.

>   * Positively verified against ``Works with Google`` certification tests. 

Reviewer: ```suggestion
  * Positively verified against "Works with Google" certification tests. 
```
It's not a code name or parameter, so no point in using ``.

* :ref:`matter_light_switch_sample`:

  * Added Wi-Fi low power configuration using Wi-Fi's :ref:`Legacy Power Save mode <ug_nrf70_developing_powersave_dtim_unicast>`.

* :ref:`matter_light_bulb_sample`:

>   * Positively verified against ``Works with Google`` certification tests.

Reviewer: ```suggestion
  * Positively verified against "Works with Google" certification tests.
```
Idem.

>   * Tested compatibility with:
>     * Google Home ecosystem for both Matter over Thread and Matter over Wi-Fi solutions (tested with Google Nest Hub 2nd generation for software version: 47.9.4.447810048 and Chromecast firmware version: 1.56.324896, and Google Home mobile application 2.63.1.12 version).

Reviewer: ```suggestion
  * Tested compatibility with the following ecosystems:
  
    * Google Home ecosystem for both Matter over Thread and Matter over Wi-Fi solutions.
       Tested with Google Nest Hub 2nd generation (software version: 47.9.4.447810048; Chromecast firmware version: 1.56.324896, and Google Home mobile application v2.63.1.12).
```

>     * Apple Home ecosystem for both Matter over Thread and Matter over Wi-Fi solutions (tested with Apple HomePod mini and Apple iPhone for iOS 16.3 version).

Reviewer 2: ```suggestion
    * Apple Home ecosystem for both Matter over Thread and Matter over Wi-Fi solutions (tested with Apple HomePod mini and Apple iPhone with iOS 16.3 version).
```

>     * Apple Home ecosystem for both Matter over Thread and Matter over Wi-Fi solutions (tested with Apple HomePod mini and Apple iPhone for iOS 16.3 version).

Reviewer: ```suggestion
    * Apple Home ecosystem for both Matter over Thread and Matter over Wi-Fi solutions.
      Tested with Apple HomePod mini and Apple iPhone (iOS v16.3).
```

>     * Samsung SmartThings ecosystem for Matter over Thread solution (tested with Aeotec Smart Home Hub and SmartThings mobile application 1.7.97.22 version).

Reviewer: ```suggestion
    * Samsung SmartThings ecosystem for Matter over Thread solution.
      Tested with Aeotec Smart Home Hub and SmartThings mobile application (v1.7.97.22).
```

>     * Amazon Alexa ecosystem for both Matter over Thread and Matter over Wi-Fi solutions (tested with Amazon Echo Dot and Amazon Alexa mobile application 2.2.495949.0 version).

Reviewer: ```suggestion
    * Amazon Alexa ecosystem for both Matter over Thread and Matter over Wi-Fi solutions.
      Tested with Amazon Echo Dot and Amazon Alexa mobile application (v2.2.495949.0).
```
